AsyncEnumerator.cs

```diff
@@ -231,7 +234,10 @@ public void Dispose()
         private void ClearState()
         {
             if (_yield != null)
+            {
+                //AsyncEnumerationCanceledException
                 _yield.Finilize();
```

Comment:
The name `Finalize` will interefere with the `Object.Finalize` method. What about just removing `Finilize` and calling `SetCanceled` directly from `Yield.ClearState`?